src/shared/ui/common/location.ts

```diff
@@ -7,91 +7,89 @@ import * as vscode from 'vscode'
 import { PrompterButtons } from '../buttons'
 
 const localize = nls.loadMessageBundle()
 
-interface Folder {
-    readonly uri: vscode.Uri
-    readonly name: string
-}
+function createBrowseFolderQuickPickItem(
+    label: string,
+    detail: string,
+    defaultUri: vscode.Uri = vscode.Uri.file(os.homedir())
+): DataQuickPickItem<vscode.Uri> {
+    const openFileDialog = async () => {
+        const result = await vscode.window.showOpenDialog({
+            defaultUri: defaultUri,
+            openLabel: localize('AWS.generic.open', 'Open'),
+            canSelectFiles: false,
+            canSelectFolders: true,
+            canSelectMany: false,
+        })
 
-class FolderQuickPickItem implements Omit<DataQuickPickItem<vscode.Uri>, 'disabled'> {
-    public readonly label: string
-    public readonly data: vscode.Uri
+        if (!result || !result.length) {
+            return WIZARD_RETRY
+        }
 
-    public constructor(folder: Folder | vscode.WorkspaceFolder) {
-        this.label = addCodiconToString('root-folder-opened', folder.name)
-        this.data = folder.uri
+        return result[0]
     }
-}
-
-class BrowseFolderQuickPickItem implements Omit<DataQuickPickItem<vscode.Uri>, 'disabled'> {
-    public alwaysShow: boolean = true
-    public skipEstimate: boolean = true
-
-    public constructor(
-        public readonly label: string,
-        public readonly detail: string,
-        private readonly defaultUri: vscode.Uri = vscode.Uri.file(os.homedir())
-    ) {}
-
-    public get data(): () => Promise<vscode.Uri | WizardControl> {
-        return async () => {
-            const result = await vscode.window.showOpenDialog({
-                defaultUri: this.defaultUri,
-                openLabel: localize('AWS.generic.open', 'Open'),
-                canSelectFiles: false,
-                canSelectFolders: true,
-                canSelectMany: false,
-            })
-
-            if (!result || !result.length) {
-                return WIZARD_RETRY
-            }
 
-            return result[0]
-        }
+    return {
+        label,
+        detail,
+        alwaysShow: true,
+        skipEstimate: true,
+        data: openFileDialog,
     }
 }
 
-interface LocationPrompterOptions {
+interface LocationPromptOptions {
+    /** Title of the prompt (default: 'Select a folder') */
     title?: string
+    /** Optionally allow the user to select any directory, shown as a new item with specificed detail */
+    browseFolderDetail?: string
+    /** Removes the option to choose any folder */
+    disableBrowseFolder?: boolean
+    /** Extra buttons to add */
     buttons?: PrompterButtons<vscode.Uri>
-    newFolderDetail?: string
 }
 
-export function createLocationPrompt(
-    folders: Folder[] | readonly vscode.WorkspaceFolder[] = [],
-    options: LocationPrompterOptions = {}
+// We only care about the name and URI of the folder
+type Folder = Pick<vscode.WorkspaceFolder, 'name' | 'uri'>
```

Comment:
Mmm perhaps. I'm in favor of having the doc strings in-lined with the interface since they show up on the properties when iterating over them with auto-complete.